pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -63,3 +65,26 @@ func WithSecretHashAnnotationHook(
 		return nil
 	}
 }
+
+// WithReplicasHook sets the deployment.Spec.Replicas field according to the number
+// of available nodes. If the number of available nodes is bigger than one, then the
+// number of replicas will be two. The number of nodes is determined by the node
+// selector specified in the field deployment.Spec.Templates.NodeSelector.
+// When node ports or hostNetwork are used, maxSurge=0 should be set in the
+// Deployment RollingUpdate strategy to prevent the new pod from getting stuck
+// waiting for a node with free ports.
+func WithReplicasHook(nodeLister corev1listers.NodeLister) DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		nodeSelector := deployment.Spec.Template.Spec.NodeSelector
+		nodes, err := nodeLister.List(labels.SelectorFromSet(nodeSelector))
+		if err != nil {
+			return err
+		}
+		replicas := int32(1)
+		if len(nodes) > 1 {
+			replicas = int32(2)
+		}
+		deployment.Spec.Replicas = &replicas
+		return nil
```

Comment:
It would be nice to have a common library-go function to get expected nr. of controller-like replicas. In this PR it's "2", but  e.g. in https://github.com/openshift/cluster-csi-snapshot-controller-operator/pull/88 it's all masters (="3").

Reply:
I can do that, but I don't have a good place in mind (not all controllers should be **2**).

The number of replicas should be evaluated based on the Deployment at hand. For instance, CSI Deployments should be **2**, but scheduler and KCM should be the number of master nodes. Also, IMO vsphere-problem-detector should be **1** because a small downtime isn't critical to the cluster.